src/lib/PriorityOrderLib.sol

```diff
@@ -24,18 +29,33 @@ struct PriorityOutput {
 struct PriorityOrder {
     // generic order information
     OrderInfo info;
+    // The address which must cosign the full order
+    address cosigner;
     // the block at which the order becomes active
-    uint256 startBlock;
+    uint256 auctionStartBlock;
+    // the block at which the order can be executed without a cosigner
+    uint256 openAuctionStartBlock;
```

Comment:
"open" is a little overloaded from v1/v2 wrt non-exclusivity - maybe "userAuctionStartBlock"?

Reply:
thinking about it more, do we actually need `auctionStartBlock` above? Seems what's happening here is:
- User sets an auction start block
- Cosigner can make it _earlier_ if they want
- is there even any risk of cosigner making it _too_ early? it doesn't really affect anything whether it's 1 second in the past or 1 year in the past afaict

Reply:
mmm so in practice, no risk of the cosigner making it too early. However an important note is that all of these values are in the future relative to the user signing the order, like auctionStartBlock is a few blocks in the future, preventing the cosigner from starting the order immediately.

Reply:
Ok I think I agree with you that we don't need auctionStartBlock. The ideal situation UX wise is to give the cosigner full control to set the target block whenever during the order lifecycle. However we want to bake in a period where the order can be submitted without the cosigner - so the question is where should this be? At the start of the order or at the end near the deadline?

Given that we are working with fuzzy numbers (buffer from frontend), I think its easiest to give ourselves more leeway here and have the "userAuctionStartBlock" be the parameter that is signed over by the user which is going to be pretty close to the deadline of the order. And then the cosigner can start the auction at any earlier block. As long as (userAuctionStartBlock - currentBlock) is enough time (doesn't have to be super long) this should be fine. 

The worry that the cosigner could start the order instantly is valid but something I think we just have to live with, it is not impossible to think that searchers could compete in < 2 seconds

Reply:
Huddled with @zhongeric on this exact topic too, and I think it comes down to whether or not we want to make orders fillable without cosigner data?

Just to recap how we got here:
- We are concerned that users can sit on the permit signature request for an unknown amount of time before signing, making the timestamp in the original request stale. This alone wouldn't break the system as we can just set deadline super loose.
  - However, we also want a small buffer in between when order is submitted, and when order becomes fillable, to allow order dissemination to enough fillers
- To combat this, we want to introduce a second timestamp which is later in time. We call it `auctionStartBlock`, and introduces a Lab-controlled cosigner that has the ability to set it to delay the auction start at the time of order submission
- However, introducing a cosigner means a new centralized entity is needed to run the auction, unless we introduce a third timestamp, `openAuctionStartBlock`, after which the cosigner is not required

I think I agree that we should design a system where Lab infra isn't  a necessary component, although I feel like we can set `openAuctionStartTime` tighter as opposed to be close to deadline to allow for faster fills in this edge case we are discussing

Reply:
Re. our sync today:
- user signs on `auctionStartBlock`
- cosigner can override to decrease this by any amount
- order is valid _only_ after the resolved `auctionStartBlock`
- thus, order is valid after the min of (auctionTargetBlock, auctionStartBlock) until deadline